tests/modules/restful_test/src/Plugin/resource/entity_test/main/v1/Main__1_8.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\restful_test\Plugin\resource\entity_test\main\v1\Main__1_8.
+ */
+
+namespace Drupal\restful_test\Plugin\resource\entity_test\main\v1;
+
+use Drupal\restful\Plugin\resource\DataInterpreter\DataInterpreterInterface;
+
+/**
+ * Class Main__1_8.
+ *
+ * @package Drupal\restful_test\Plugin\resource
+ *
+ * @Resource(
+ *   name = "main:1.8",
+ *   resource = "main",
+ *   label = "Main",
+ *   description = "Export the entity test 'main' bundle.",
+ *   authenticationOptional = TRUE,
+ *   dataProvider = {
+ *     "entityType": "entity_test",
+ *     "bundles": {
+ *       "main"
+ *     },
+ *   },
+ *   majorVersion = 1,
+ *   minorVersion = 8
+ * )
+ */
+class Main__1_8 extends Main__1_0 {
+
+  /**
+   * Overrides ResourceEntity::publicFields().
+   */
+  protected function publicFields() {
+    $public_fields = parent::publicFields();
+
+    $class = get_class($this);
+    $public_fields['random_rel'] = array(
```

Comment:
random is probably a misleading name. arbitrary? or `property_reference`?

Anyway, if I understand correctly this could be used for example on `$node->uid` to convert the response to a full blown user resource?